src/test/java/picard/fingerprint/FingerprintCheckerTest.java

```diff
@@ -316,8 +320,45 @@ public void testQueryable(final File vcf, boolean expectedQueryable) {
         }
     }
 
+    @DataProvider
+    Object[][] deepDataProvider() {
+        return new Object[][]{{10}, {50}, {100}, {500}, {1000}, {5000}, {10000}, {50000}};
+    }
+
+    @Test(dataProvider = "deepDataProvider")
+    void testCanHandleDeepData(int counts) {
+        final HaplotypeProbabilitiesFromSequence hp1 = new HaplotypeProbabilitiesFromSequence(hb);
+        addObservation(hp1, hb, counts, hb.getFirstSnp().getAllele1());
+        addObservation(hp1, hb, counts, hb.getFirstSnp().getAllele2());
+
+        final double[] logLikelihoods = hp1.getLogLikelihoods();
+        Assert.assertTrue( MathUtil.min(logLikelihoods) < 0 );
+    }
+
+    @Test(dataProvider = "deepDataProvider")
+    void testCanHandleDeepDataHPFromContamination(int counts) throws IOException {
+        final double maf = 0.4;
+        final Snp snp = new Snp("test", "chr1", 1, (byte) 'A', (byte) 'T', maf, Collections.singletonList("dummy"));
+        final HaplotypeBlock hb = new HaplotypeBlock(maf);
+        hb.addSnp(snp);
+
+        final HaplotypeProbabilitiesFromSequence hp1 = new HaplotypeProbabilitiesFromContaminatorSequence(hb,.999);
+        addObservation(hp1, hb, counts, hb.getFirstSnp().getAllele1());
+        addObservation(hp1, hb, counts, hb.getFirstSnp().getAllele2());
+
+        final double[] logLikelihoods = hp1.getLogLikelihoods();
+        Assert.assertTrue( MathUtil.min(logLikelihoods) < 0);
```

Comment:
is there any reason not to check that the logLikelihoods match the actual values you expect, instead of just checking that they haven't underflowed?

Reply:
With a data provider making this many tests, I didn't feel like calculating the expectation and this was enough to catch the underflow, which was the purpose of this PR...